cmd/noms/splore/noms_splore.go

```diff
@@ -0,0 +1,307 @@
+		//http.ServeFile(w, r, "splore/out.js")
+		// To build noms-splore. uncomment this line and run `yarn buildgo`:
+		fmt.Fprint(w, outJs)
+	})
+
+	mux.HandleFunc("/getNode", func(w http.ResponseWriter, r *http.Request) {
+		req(w, "application/json")
+		r.ParseForm()
+		id := r.Form.Get("id")
+
+		var v types.Value
+		switch {
+		case id == "":
+			v = getValue()
+		case id[0] == '#':
+			abspath, err := spec.NewAbsolutePath(id)
+			d.PanicIfError(err)
+			v = abspath.Resolve(sp.GetDatabase())
+		default:
+			path := types.MustParsePath(id)
+			v = path.Resolve(getValue(), sp.GetDatabase())
+		}
+
+		if v == nil {
+			http.Error(w, `{"error": "not found"}`, http.StatusNotFound)
+			return
+		}
+
+		err := json.NewEncoder(w).Encode(node{
+			nodeInfo: info(v, id),
+			Children: getNodeChildren(v, id),
+		})
+		d.PanicIfError(err)
+	})
+
+	l, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	d.PanicIfError(err)
+
+	if browse {
+		open.Run("http://" + l.Addr().String())
+	}
+
+	if !verbose.Quiet() {
+		fmt.Println("Listening on", l.Addr().String())
+	}
+
+	d.PanicIfError(httpServe(l, mux))
+	return 0
+}
+
+func getNodeChildren(v types.Value, parentPath string) (children []nodeChild) {
+	atPath := func(i int, suffix string) string {
+		return fmt.Sprintf("%s@at(%d)%s", parentPath, i, suffix)
+	}
+
+	switch v := v.(type) {
+	case types.Bool, types.Number, types.String:
+		children = []nodeChild{}
+	case types.Blob:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = []nodeChild{}
+		}
+	case types.List:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = make([]nodeChild, v.Len())
+			v.IterAll(func(vi types.Value, i uint64) {
+				children[i] = nodeChild{
+					Value: info(vi, fmt.Sprintf("%s[%d]", parentPath, i)),
+				}
+			})
+		}
+	case types.Map:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = make([]nodeChild, v.Len())
+			i := 0
+			v.IterAll(func(k, v types.Value) {
+				children[i] = nodeChild{
+					Key:   info(k, atPath(i, "@key")),
+					Value: info(v, atPath(i, "")),
+				}
+				i++
+			})
+		}
+	case types.Set:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = make([]nodeChild, v.Len())
+			i := 0
+			v.IterAll(func(v types.Value) {
+				children[i] = nodeChild{
+					Value: info(v, atPath(i, "")),
+				}
+				i++
+			})
+		}
+	case types.Ref:
+		children = []nodeChild{{
+			Value: info(v, parentPath+"@target"),
+		}}
+	case types.Struct:
+		children = make([]nodeChild, v.Len())
+		i := 0
+		v.IterFields(func(name string, v types.Value) {
+			children[i] = nodeChild{
+				Label: name,
+				Value: info(v, fmt.Sprintf("%s.%s", parentPath, name)),
+			}
+			i++
+		})
+	default:
+		panic(fmt.Errorf("unsupported value type %T", v))
+	}
+	return
+}
+
+func nodeName(v types.Value) string {
+	typeName := func(iface interface{}) string {
+		return reflect.TypeOf(iface).Name()
+	}
+
+	switch v := v.(type) {
```

Comment:
This is usually done by checking `Kind()`. You can also do `Kind().String()` to get the human readable name of the `NomsKind`.